src/main/kotlin/racingcar/domain/RacingGame.kt

```diff
@@ -0,0 +1,36 @@
+package racingcar.domain
+
+import racingcar.domain.interfaces.Raceable
+
+/**
+ * 자동차 경주 게임
+ * */
+const val MIN_RANDOM_VALUE = 0
+const val MAX_RANDOM_VALUE = 9
+const val CAR_NAME_DELIMITER = ','
+
+object RacingGame {
+
+    /**
+     * 자동차 경주 시작
+     * @param racingGameInfo first: 경주 자동차 리스트, second: 라운드 수
+     * @param roundFinishListener 라운드 종료 리스너
+     * */
+    fun run(racingGameInfo: Pair<List<Raceable>, Int>, roundFinishListener: (List<Raceable>, Boolean) -> Unit) {
+
+        var (carList, roundCnt) = racingGameInfo
+
+        repeat(roundCnt) { round ->
+            carList = playRound(carList)
+            roundFinishListener(carList, round == roundCnt - 1)
+        }
+    }
+
+    /**
+     * 게임 시작
+     * @param raceableList 자동차 리스트
+     * @return 자동차 경주 한 라운드 결과를 가지고 있는 자동차 리스트
+     * */
+    private fun playRound(raceableList: List<Raceable>): List<Raceable> =
+        raceableList.map { it.move((MIN_RANDOM_VALUE..MAX_RANDOM_VALUE).random()); it }
```

Comment:
고차함수로 전진 전략에 대한 조건을 작성해주셨는데, 전략패턴(strategy pattern)을 통해 행위 추상화로 SOLID원칙을 지켜보는건 어떨까요? 

```suggestion
        raceableList.map { it.move(randomThresholdMoveStrategy); it }
```

그리고 ; 뒤에 it을 놓는건 가독성이 떨어질 수 밖에 없는 것 같네요! ㅠㅠ 
이보다는 move가 자기자신을 반환하도록 반환타입을 지정하거나 여기선 map이 아닌 forEach를 사용하고
요청 측에서 이를 고려해서 로직을 변경하도록 하는게 낫지 않을까요?!